packages/node/src/indexer/api.service.ts

```diff
@@ -72,6 +72,10 @@ export class ApiService
     try {
       chainTypes = this.project.chainTypes;
       network = this.project.network;
+
+      if (this.nodeConfig.primaryNetworkEndpoint) {
+        network.endpoint.push(this.nodeConfig.primaryNetworkEndpoint);
```

Comment:
How does connection pool service know this is a primary endpoint?

Reply:
The check happens here:

https://github.com/subquery/subql/blob/6e57956ffebc52f7a97255edf519c1688b8fb10a/packages/node-core/src/indexer/connectionPool.service.ts#L54